python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -304,3 +303,12 @@ def _update_common_interface_cfg(
             if p2p_link.include_in_underlay_protocol is True and self.shared_utils.underlay_ldp and default(p2p_link.mpls_ldp, True):  # noqa: FBT003
                 interface.mpls.ldp.interface = True
                 interface.mpls.ldp.igp_sync = True
+
+    def _get_channel_id(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol, p2p_link: T_P2pLinksItem, node_data: dict) -> int:
+        """Returns a channel ID for one p2p_link."""
+        if p2p_link.port_channel.channel_id_generation == "p2p_link_id":
+            default_channel_id = p2p_link.id + p2p_link.port_channel._get("channel_id_offset", 0)
```

Comment:
What happens if I don't give an ID on the p2p_link?

Reply:
Added check for this and corresponding unit test.